### ISSUE TITLE
Selection ranges for some more complex expressions does not round-trip properly (reading /setting produces a different selection) 

### ISSUE THREAD

**Issue by mindreframer** [arnog/mathlive#902]
### BUG: 
Selection ranges for some more complex expressions do not round-trip properly (reading /setting produces a different selection) 

## Steps to Reproduce
- Repo with a reproduction: https://github.com/mindreframer/mathlive-selection-bug 
- Screen-recording: https://www.loom.com/share/70f3871eb9974cbaae056160d51f0491

### Actual Behavior
Mathlive is not calculating the selection correctly and the range grows after setting it: `1-6 -> 1-8, 1-12 -> 1-33`

### Expected Behavior
Setting the selection should result in that same selection when reading back. 

### Environment

- Mac OSX / 10.15.7 (Catalina) 
- Chrome Version 89.0.4389.114 (Official Build)

**Comment by mindreframer** [comment]
@arnog After updating to mathlive 0.60.1, the reading and re-setting the read selection produces correct results, at least in the couple cases that I threw at it... Sorry, was not aware that there was a new release for Mathlive, that also had substantial changes for the selection handling. 

I guess this issue is fixed then :) Thanks for the new release!

**Comment by arnog** [comment]
@mindreframer That's great news!

There could certainly still be some bugs lurking around. This is a new area of the code and they were a lot of changes needed to implement it. 

Don't hesitate to file issues. 

Also, thank you *so much* for preparing a repo with demonstrated the problem. This is invaluable for debugging and will often be the difference between a bug that I can fix promptly and one that lingers or that I have to close as "could not reproduce".

**Comment by mindreframer** [comment]
@arnog I will make sure to report new issues as I find them 😅 

The repo: it was provided it for purely selfish reasons to avoid explaining the issue with long, hard to understand sentences or locally not reproducible screenshots 😄  As a fellow developer the best way to prove your point is to provide the real thing, means code (working or not working). 

Another idea that might make reported issues more actionable: 
- we could provide 2-3 scaffolded repos with Mathlive configured in different ways. Like: 
  - static precompiled lib with HTML
  - NPM lib with React / Vue / pure Mathlive (JS / TS) 
To keep things simple, it could be a single mono-repo with YARN and packages for each variation. 

OR: it could be a modified version for a create-app package, like this: 
- `https://github.com/vitejs/vite/tree/main/packages/create-app`

On bug issues the reporter is encouraged to fork / use this repo and reproduce the issue herself. The issue will get a label "has-reproduction". 

For other issues, that are harder to reproduce, the reporter is encouraged to record a short screencast with `loom.com`, the free version allows screencasts upto 5 minutes and the software is the least glitchy from all the browser-based screenrecording tools that I have tested (and I have tested a lot of them). 

Because there is an officially "blessed" recommendation, the dilemma of making a choice how to report a bug is reduced and people tend to stick to the recommendations to increase the likelihood of fixing their reported issue.


I can take a stub at providing this repo. I have a feeling I might be using it myself often in the future, because the current application  that I'm building is using Mathlive as a core component.
